abcclassroom/clone.py

```diff
@@ -63,30 +90,46 @@ def clone_student_repos(args):
 
     if materials_dir is None:
         print(
-            "No course_materials directory set in config.yml. Will clone "
-            "repositories but will not copy any assignment files."
+            "Oops! I couldn't find a course_materials directory location "
+            "in your config.yml file. I will just clone all of the student"
+            "repositories. I can't copy any assignment files to a "
+            "course_materials directory."
         )
+    # This is a really large try block and makes it hard to figure out where
+    # things fail. perhaps break out into a few sub try / excepts?
     try:
         # Create the assignment subdirectory path and ensure it exists
         Path(course_dir, clone_dir, assignment_name).mkdir(exist_ok=True)
         missing = []
+        # If it can't find the roster file you'll get a FileNotFoundError
+        # Also what happens if the roster is not in the correct format??
         with open(roster_filename, newline="") as csvfile:
             reader = csv.DictReader(csvfile)
-            for row in reader:
-                student = row["github_username"]
-                # expected columns: identifier,github_username,github_id,name
-                repo = "{}-{}".format(assignment_name, student)
-                try:
-                    clone_or_update_repo(
-                        organization,
-                        repo,
-                        Path(clone_dir, assignment_name),
-                        skip_existing,
-                    )
-                    if materials_dir is not None:
-                        copy_assignment_files(config, student, assignment_name)
-                except RuntimeError:
-                    missing.append(repo)
+            try:
+                for row in reader:
```

Comment:
this is how we'd check the headers are correct:

            # Get headers and check that gh username is there?
            if not "github_username" in reader.fieldnames

@kcranston this is going to add some additional non ducktyping approach tests to this function if we implement it. should we add that to check the headers? or just ask for forgiveness here as i setup?